atlasdb-config/src/main/java/com/palantir/atlasdb/http/AtlasDbHttpProtocolVersion.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.http;
+
+import java.util.List;
+
+import javax.ws.rs.core.HttpHeaders;
+
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.conjure.java.api.config.service.UserAgents;
+
+public enum AtlasDbHttpProtocolVersion {
+    LEGACY_OR_UNKNOWN,
+    CONJURE_JAVA_RUNTIME;
+
+    public static AtlasDbHttpProtocolVersion inferFromHttpHeaders(HttpHeaders headers) {
```

Comment:
talked about this in #4192

Reply:
Yeah, the abstract class model would probably work better - as discussed on that PR. I'll do that refactoring in 4192.